iXBRLViewerPlugin/viewer/src/js/search.js

```diff
@@ -108,10 +109,11 @@ export class ReportSearch {
 
         rr.forEach((r,i) => {
                 var item = searchIndex._report.getItemById(r.ref);
-                if (
-                    (item.isHidden() ? s.showHiddenFacts : s.showVisibleFacts) &&
+                if ((item.isHidden() ? s.showHiddenFacts : s.showVisibleFacts) &&
                     (s.periodFilter == '*' || item.period().key() == s.periodFilter) &&
-                    (s.conceptTypeFilter == '*' || s.conceptTypeFilter == (item.isNumeric() ? 'numeric' : 'text'))) {
+                    (s.conceptTypeFilter == '*' || s.conceptTypeFilter == (item.isNumeric() ? 'numeric' : 'text')) &&
+                    (s.factValueFilter == '*') || (item.isNumeric() && !item.isNil() && s.factValueFilter == (new Decimal(item.value()).isPositive() ? 'positive': 'negative')))
```

Comment:
The parentheses look wrong; I think it needs to be `(s.factValueFilter == '*' || ...)` or else the value filter won't intersect with the other filters.  I don't see any pre-existing tests for search, but it seems like that would be good to add at some point.

Is `new Decimal` correct, or should it be `Decimal` (as in `formatNumber`)?  Do we need an `undefined` test?  Perhaps we should add a helper method to fact to do this.  You might consider naming the method `isNegative()` since `isPositive` returning true for 0 seems somewhat confusing to me.

Reply:
> The parentheses look wrong; I think it needs to be (s.factValueFilter == '*' || ...) or else the value filter won't intersect with the other filters

Good catch, thanks

> I don't see any pre-existing tests for search, but it seems like that would be good to add at some point.

I can attempt this

> Is new Decimal correct, or should it be Decimal (as in formatNumber)?

I looked at `formatNumber` and it will return a bunch of things that aren't actually a decimal which is why I decided to just check the conditions for being a valid `Decimal`  being returned upfront and pass it in directly so that I am guaranteed to have access to `isPositive()`

>  Do we need an undefined test?

Maybe? I don't know what that explicitly refers to though

> Perhaps we should add a helper method to fact to do this.

Seems reasonable to me.

> You might consider naming the method isNegative()

Roger that.

Reply:
I didn't resolve the fact helper question mostly because I didn't know what it would look like ideally

Reply:
> You might consider naming the method isNegative() since isPositive returning true for 0 seems somewhat confusing to me.

I agree that the method should be named precisely, but what is the desired end user behaviour here?  If a user selects "positive" in the filter, do they want zeros included?  I'm guessing the use case here is for reviewing facts that have been tagged with the wrong sign, in which case you probably want to ignore zeros for both "positive" and "negative" filters.

FWIW, XML Schema uses "nonNegative" and "nonPositive" in datatype names order to unambiguously include zeros.